server/src/main/java/org/opensearch/search/aggregations/metrics/CardinalityAggregator.java

```diff
@@ -212,6 +287,125 @@ private abstract static class Collector extends LeafBucketCollector implements R
 
     }
 
+    private static class PruningCollector extends Collector {
+
+        private final Collector delegate;
+        private final DisiPriorityQueue queue;
+        private final DocIdSetIterator competitiveIterator;
+
+        PruningCollector(Collector delegate, TermsEnum terms, LeafReaderContext ctx, SearchContext context, String field)
+            throws IOException {
+            this.delegate = delegate;
+
+            Map<BytesRef, Scorer> postingMap = new HashMap<>();
+            while (terms.next() != null) {
+                BytesRef term = terms.term();
+                TermQuery termQuery = new TermQuery(new Term(field, term));
+                Weight subWeight = context.searcher().createWeight(termQuery, ScoreMode.COMPLETE_NO_SCORES, 1f);
```

Comment:
please change it to -                 
```
Weight subWeight = termQuery.createWeight(context.searcher(), ScoreMode.COMPLETE_NO_SCORES, 1f);
```
When we use `context.searcher().createWeight()` it uses OpenSearch `ContextIndexSearcher` and starts using bunch of wrapped functionality which we don't need here. For example, if profile is set, it will initialize all sorts of profile related object which aren't required.

Reply:
Aha, that's why I saw a lots of queries in the profile results.